scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -549,6 +554,46 @@ def add_testsuites(self, testsuite_filter=None):
                                 suite.platform_allow,
                                 f"platform_allow in {suite.name}")
 
+                        if self.options.log_level == "QA":
+                            if len(suite.platform_allow) > 10:
+                                logger.qa(f"Too many entries in allow_platform in {suite.name}: {len(suite.platform_allow)}")
+
+                            if len(suite.integration_platforms) > 5:
+                                logger.qa(f"Too many integration platforms in {suite.name}: {len(suite.integration_platforms)}")
+
+                            if len(suite.platform_exclude) > 5:
+                                logger.qa(f"Too many excluded platforms in {suite.name}: {len(suite.platform_exclude)}")
+
+                            if len(suite.platform_allow) > 1 and not suite.integration_platforms:
+                                logger.qa(f"platform_allow in {suite.name} without integration_platforms")
+
+
+                            if suite.build_only and not "build" in suite.id:
+                                logger.qa(f"build_only set in {suite.name}")
+
+                            if suite.skip:
+                                logger.qa(f"skip set in {suite.name}")
+
+                            if suite.harness not in SUPPORTED_HARNESS:
+                                logger.qa(f"Unsupported harness {suite.harness} in {suite.name}")
+
+                            if suite.filter and not suite.integration_platforms and not suite.platform_allow:
+                                logger.qa(f"filter with no integration platforms in {suite.name}")
+
+                            if suite.platform_allow and suite.integration_platforms:
+                                _default_p = set(self.default_platforms)
+                                _platform_allow = set(suite.platform_allow)
+                                _integration_p = set(suite.integration_platforms)
+                                _intersection1 = _default_p.intersection(_platform_allow)
+                                _intersection2 = _default_p.intersection(_integration_p)
+
+                                if  _intersection1 and not _intersection2:
+                                    logger.qa(f"No default platforms in integration_platform for {suite.name}")
+
+                            _do = set(suite.depends_on).intersection(self.supported_features)
+                            if len(_do) != len(suite.depends_on):
+                                logger.qa(f"Unsupported depends_on {suite.depends_on} in {suite.name}")
```

Comment:
IMO we would have better control and understanding if we extract all existing supported_features and depends_on in advance and add them to constants module instead of recreating the list of supported_features each time. Then we will always see when something new/wrong is being added.

The logic could work both way: having depends_on which no platform supports, but also platform having a supported key which is not used in any test as depends_on

Reply:
if we want to put this in a constant is open for discussion, right now this tries to use the data as we have it.